SearchingEngine.java

```diff
@@ -0,0 +1,167 @@
+                    peopleMap.put(str.toLowerCase(), integerList);
+                } else {
+                    currentWordIndexesList.add(i);
+                    peopleMap.put(str.toLowerCase(), currentWordIndexesList);
+                }
+            }
+            i++;
+        }
+
+
+        switch (STRATEGY) {
+            case "ALL":
+
+                for (String value : query) {
+                    if (peopleMap.containsKey(value)) {
+                        index.addAll(peopleMap.get(value));
+                    }
+                }
+                Set<String> sR = new HashSet<>();
+                for (Integer integer : index) {
+                    sR.add(allList.get(integer));
+                }
+                for (String s : sR) {
+                    if ((Arrays.asList(s.split(" "))).containsAll(Arrays.asList(query))) {
+                        searchResult.add(s);
+                    }
+                }
+
+                if (!searchResult.isEmpty()) {
+                    System.out.printf("%d persons found: \n", searchResult.size());
+                    searchResult.forEach(System.out::println);
+                } else {
+                    System.out.println("No matching people found!");
+                }
+                break;
+
+            case "ANY":
+
+                for (String value : query) {
+                    if (peopleMap.containsKey(value)) {
+                        index.addAll(peopleMap.get(value));
+                    }
+                }
+                for (Integer integer : index) {
+                    searchResult.add(allList.get(integer));
+                }
+                if (!searchResult.isEmpty()) {
+                    System.out.printf("%d persons found: \n", searchResult.size());
+                    searchResult.forEach(System.out::println);
+                } else {
+                    System.out.println("No matching people found!");
+                }
+                break;
+
+            case "NONE":
+
+                for (String value : query) {
+                    if (peopleMap.containsKey(value)) {
+                        index.addAll(peopleMap.get(value));
+                    }
+                }
+                for (Integer integer : index) {
+                    searchResult.add(allList.get(integer));
+                }
+                allList.removeAll(searchResult);
+                if (!allList.isEmpty()) {
+                    System.out.printf("%d persons found: \n", allList.size());
+                    allList.forEach(System.out::println);
+                } else {
+                    System.out.println("No matching people found!");
+                }
+                break;
+        }
+    }
+
+    public static void printAll(List<String> allList) {
+        System.out.println("=== List of people==");
+        allList.forEach(System.out::println);
+    }
+
+    public static String enterStringData(String message) {
```

Comment:
Some of these methods can be made private, as they act as helper method to this class.